Tools/ardupilotwaf/boards.py

```diff
@@ -5,232 +5,231 @@
-        '-O3',
-    ]
-
-    env.LIB += [
-        'm',
-    ]
-    env.LINKFLAGS += ['-pthread',]
-    env.AP_LIBRARIES += [
-        'AP_HAL_SITL',
-        'SITL',
-    ]
-
-define_board(sitl, 'sitl')
-
-
-def linux(env):
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD = 'HAL_BOARD_LINUX',
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_NONE',
-    )
-
-    env.CXXFLAGS += [
-        '-O3',
-    ]
-
-    env.LIB += [
-        'm',
-        'rt',
-    ]
-    env.LINKFLAGS += ['-pthread',]
-    env.AP_LIBRARIES = [
-        'AP_HAL_Linux',
-    ]
-
-define_board(linux, 'linux')
-
-
-def minlure(env):
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_MINLURE',
-    )
-
-define_board(minlure, 'minlure', 'linux')
-
-
-def erleboard(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_ERLEBOARD',
-    )
-
-define_board(erleboard, 'erleboard', 'linux')
-
-
-def navio(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_NAVIO',
-    )
-
-define_board(navio, 'navio', 'linux')
-
-
-def zynq(env):
-    env.TOOLCHAIN = 'arm-xilinx-linux-gnueabi'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_ZYNQ',
-    )
-
-define_board(zynq, 'zynq', 'linux')
-
-
-def bbbmini(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_BBBMINI',
-    )
-
-define_board(bbbmini, 'bbbmini', 'linux')
-
-
-def pxf(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_PXF',
-    )
-
-define_board(pxf, 'pxf', 'linux')
-
-
-def bebop(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_BEBOP',
-    )
-
-    env.STATIC_LINKING = [True]
-
-define_board(bebop, 'bebop', 'linux')
-
-
-def raspilot(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_RASPILOT',
-    )
-
-define_board(raspilot, 'raspilot', 'linux')
-
-
-def erlebrain2(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_ERLEBRAIN2',
-    )
-
-define_board(erlebrain2, 'erlebrain2', 'linux')
-
-
-def bhat(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_BH',
-    )
-
-define_board(bhat, 'bhat', 'linux')
-
-
-def pxfmini(env):
-    env.TOOLCHAIN = 'arm-linux-gnueabihf'
-
-    env.DEFINES.update(
-        CONFIG_HAL_BOARD_SUBTYPE = 'HAL_BOARD_SUBTYPE_LINUX_PXFMINI',
-    )
-
-define_board(pxfmini, 'pxfmini', 'linux')
+class sitl(Board):
+    def configure_env(self, env):
```

Comment:
Minor: could the content of `configure_env` implementations simply be in their `__init__` functions? Given it's unlikely we expect two calls for `configure` to give us different results, the env could be calculated only once.